### DISCUSSION TITLE
How to invoke <redirect_uri>?revoke_session=<uuid>

### DISCUSSION THREAD

**Discussion by spiazzi** [OpenIDC/mod_auth_openidc#1091] [Q&A]
Hi all, I try to figure out how to call the revoke_session via <redirect_uri>?revoke_session= uuid .

I tried to invoke via a browser with:
OIDCRedirectURI value and sessionid cookie value as uuid, but it seems it doesn't work.

Also - if I would like to send the UUID to idp to store last UUID assigned to the user, is there a way to do, so that on a next logon of the same user IDP could call <redirect_uri>?revoke_session=<uuid> to revoke the previous session used by the user and let last login only to be able to access to the protected resources?

**Comment by zandbelt** [comment]
the call should work as you describe

wrt. the IDP revoke scenario: you should not try and implement that with a non-standard extension, custom code , no protocol security review/guarantees and a fragile flow as described; if required the IDP could trigger standards-based Single Logout before authenticating the user to another application but in general I would not recommend pursuing this because it is practically inviable

**Reply by spiazzi** [reply]
Idp can login a new user , but if a session is open on openidp module, the former one seems could work anyway.
This is a common scenario to let only last one to access and revoke all other former session.

Why this is not reccomended?

During a new login on Idp , Idp could logout former session, but the idea is to call revoke session of this mod to revoke also the session from mod_auth_openidc and its cache, so that a new request do not allow former user to login. Are there other way to revoke the mod_auth_openidc session during idc logout?
